pyflagser/flagser_count.py

```diff
@@ -0,0 +1,127 @@
+        to pass an upper-triangular matrix.
+
+    Returns
+    -------
+    out : list of int
+        Cell count (number of simplices) per dimension greater than or equal
+        than `min_dimension` and less than `max_dimension`.
+
+    Notes
+    -----
+    The input graphs cannot contain self-loops, i.e. edges that start and end
+    in the same vertex, therefore diagonal elements of the input adjacency
+    matrix will be ignored.
+
+    References
+    ----------
+    .. [1] D. Luetgehetmann, "Documentation of the C++ flagser library";
+           `GitHub: luetge/flagser <https://github.com/luetge/flagser/blob/\
+           master/docs/documentation_flagser.pdf>`_.
+
+    """
+    # Extract vertices and edges
+    vertices, edges = _extract_unweighted_graph(adjacency_matrix)
+
+    # Call flagser_count binding
+    cell_count = compute_cell_count(vertices, edges, directed)
+
+    return cell_count
+
+
+def flagser_count_weighted(adjacency_matrix, max_edge_weight=None,
+                           directed=True):
+    """Compute the cell count per dimension of a directed/undirected
+    filtered flag complex.
+
+    From an adjacency matrix construct a filtered flag complex as a sequence of
+    its cells associated to their filtration values and compute the number of
+    cells per dimension at the end of the filtration.
+
+    Parameters
+    ----------
+    adjacency_matrix : 2d ndarray or scipy.sparse matrix of shape \
+        (n_vertices, n_vertices), required
+        Matrix representation of a directed/undirected weighted graph. Diagonal
+        elements are vertex weights. The way zero values are handled depends on
+        the format of the matrix. If the matrix is a dense ``numpy.ndarray``,
+        zero values denote zero-weighted edges. If the matrix is a sparse
+        ``scipy.sparse`` matrix, explicitly stored off-diagonal zeros and all
+        diagonal zeros denote zero-weighted edges. Off-diagonal values that
+        have not been explicitely stored are treated by ``scipy.sparse`` as
+        zeros but will be understood as infinitely-valued edges, i.e., edges
+        absent from the filtration.
+
+    max_edge_weight : int or float or ``None``, optional, default: ``None``
+        Maximum edge weight to be considered in the filtration. All edge
+        weights greater than that value will be considered as
+        infinitely-valued, i.e., absent from the filtration. If ``None``, all
+        finite edge weights are considered.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, computes persistent homology for the directed filtered
+        flag complex determined by `adjacency_matrix`. If False, computes
+        persistent homology for the undirected filtered flag complex obtained
+        by considering all weighted edges as undirected, and it is therefore
+        sufficient (but not necessary) to pass an upper-triangular matrix. When
```

Comment:
Given your very valid points in https://github.com/giotto-ai/giotto-tda/pull/339#discussion_r430319243 about the subtle interplay with `max_edge_weight`, I wonder if we should change the wording of parts of this here and in `flagser_weighted`.

Reply:
I have made a proposal keeping what we wrote in `gtda` but keeping the details on why this is the case. What do you think?

Reply:
It looks like `flagser_unweighted` got changed instead of `flagser_weighted` and `flagser_count_weighted`.

Reply:
I fixed it.